cpu/nrf52/spi_twi_irq.c

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (C) 2020 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     cpu_nrf52
+ * @{
+ *
+ * @file
+ * @brief       Shared IRQ handling between SPI and TWI peripherals on the nRF52
+ *              devices
+ *
+ * @author      Koen Zandberg <koen@bergzand.net>
+ *
+ * @}
+ */
+
+#include "cpu.h"
+#include "periph_cpu.h"
+
+#ifdef CPU_MODEL_NRF52840XXAA
+#define SPI_TWI_IRQ_NUMOF 4
+#else
+#define SPI_TWI_IRQ_NUMOF 3
+#endif
+
+static spi_twi_irq_cb_t _irq[SPI_TWI_IRQ_NUMOF];
+static void *_irq_arg[SPI_TWI_IRQ_NUMOF];
+
+/* I2C and SPI share peripheral addresses */
+static size_t _spi_dev2num(void *dev)
+{
+    if (dev == NRF_SPIM0) {
+        return 0;
+    }
+    else if (dev == NRF_SPIM1) {
+        return 1;
+    }
+    else if (dev == NRF_SPIM2) {
+        return 2;
+    }
+#ifdef CPU_MODEL_NRF52840XXAA
+    else if (dev == NRF_SPIM3) {
+        return 3;
+    }
+#endif
+    else {
+        assert(false);
+        return 0;
+    }
+}
+
+static const IRQn_Type _isr[] = {
+    SPIM0_SPIS0_TWIM0_TWIS0_SPI0_TWI0_IRQn,
+    SPIM1_SPIS1_TWIM1_TWIS1_SPI1_TWI1_IRQn,
```

Comment:
`TWIM0_TWIS0_TWI0_SPIM1_SPIS1_SPI1_IRQn` is indeed a thing, but @Citrullin forgot to commit the [vendor file](https://github.com/NordicSemiconductor/nrfx/blob/master/mdk/nrf52811.h) :wink:

Reply:
:scream: